c++/src/kj/compat/http.c++

```diff
@@ -2177,10 +2177,14 @@ private:
     // is empty, then they make the write directly, using `writeInProgress` to detect and block
     // concurrent writes.
 
-    writeQueue = writeQueue.then([this,content=kj::mv(content)]() mutable {
-      auto promise = inner.write(content.begin(), content.size());
-      return promise.attach(kj::mv(content));
-    });
+    constexpr auto doWrite = [](Promise<void> writeQueue,
```

Comment:
maybe better name for writeQueue? `writesBefore`?

Also I wonder if there's a better way to express such a queue rather than using promises?

Reply:
Ideally we get rid of this write queue, it's a pattern that has been known to cause trouble. Once we have `AsyncOutputStream::end()` it should be easier, as we can write the chunk terminator as part of `end()` instead of deferring it.

I would keep the name `writeQueue` for now. It's a pattern that appears in a few places.

Reply:
> Also I wonder if there's a better way to express such a queue rather than using promises?

I think we do need a general alternative, but I haven't been able to think one up in the past.